lib/depends.ps1

```diff
@@ -21,13 +21,13 @@ function deps($app, $arch) {
 
 function dep_resolve($app, $arch, $resolved, $unresolved) {
     $app, $bucket, $null = parse_app $app
+    if(((Get-LocalBucket) -notcontains $bucket) -and $bucket) {
```

Comment:
Excahange these two conditions.
In case there is no $bucket is not specified, then it will fail immediatelly and do not check the second condition. (Short circuiting)

```powershell
function testBucket($buck) { Write-Host $buck; return $true }
```

![image](https://user-images.githubusercontent.com/13260377/68201380-f8d52500-ffc1-11e9-9866-4b10e7207659.png)

See image above. Function is not evaluated in case first condition is false. And in second you can see `ahoj` message popuped from evaluation

```suggestion
    if ($bucket -and ((Get-LocalBucket) -notcontains $bucket)) {
```